server/api/server.go

```diff
@@ -40,6 +40,8 @@ type Server struct {
 	// this mutex to avoid data races.
 	mtx      sync.Mutex
 	listener net.Listener
+
+	ListenerAddr string
```

Comment:
Is there a way to do that without adding something in the exposed structs, functions?
This feels like polluting existing functions only to solve the testutil/network.
Cannot you simply dial the port, when using Linux, and if it is open, use another one?

Reply:
The thing is that between the time we get a free port and the time it gets used, some other thing can get the same port. Which is what's happening now. So we need a way to get the port once it's running. I'm still experimenting tho

Reply:
Yes, but what if we allocate a bunch of free ports ahead of time (i.e. a global pool) and then distribute ports from that pool. Then you don't have this race condition.

Reply:
Just pushed an improvement (not a solution tho).
I'll explore the global pool idea next, although Idk how that will look as we are spinning many networks in many tests that I'm not sure they run under the same process 🤔 Maybe a slice with a mutex is enough.

Reply:
What do you mean not the same process? Each Go test invocation is from the same root process. There's already a synchronized pool for this -> https://go.dev/src/sync/pool.go

Reply:
I misread a comment, will try that, should work just fine 🤞